### PR TITLE
Create Rhevm Config Subclass

### PR DESCRIPTION
**Pull request by jirihnidek**
* Created RHEVM config section subclass
* Fixed all unit tests in: test_rhevm.py
* Tested with RHEVM server
* Removed obsolete and unused testing code in __name__ == '__main__'
* Attempt to fix some problems in complex unit tests

### CODE REVIEW

**Comment by jirihnidek** [comment]
[[8] [CONFIG_EPIC] Create Rhevm Config Subclass](https://trello.com/c/NvDfv97y/583-8-configepic-create-rhevm-config-subclass)

File: virtwho/virt/rhevm/rhevm.py [hunk 1]
```diff
@@ -50,6 +49,47 @@
...
+    This class is used for validation of RHEVM virtualization backend
+    section. It tries to validate options and combination of options that
+    are specific for this virtualization backend.
+    """
+
+    VIRT_TYPE = 'rhevm'
+
+    def __init__(self, section_name, wrapper, *args, **kwargs):
+        super(RhevmConfigSection, self).__init__(section_name, wrapper, *args, **kwargs)
+
```
Comment (cnsnyder) [inline virtwho/virt/rhevm/rhevm.py]: I think the following is necessary to ensure 'server', 'username' and 'password' (or the encrypted variety) is provided:
```
        self.add_key('server', validation_method=self._validate_server, required=True)
        self.add_key('username', validation_method=self._validate_username, required=True)
        self.add_key('password', validation_method=self._validate_unencrypted_password, required=True)
```

**Review by cnsnyder — CHANGES_REQUESTED** [review]
I believe just a few small changes necessary for complete validation.

Other than the above, looks good.

File: virtwho/virt/rhevm/rhevm.py [hunk 2]
```diff
@@ -199,20 +229,20 @@ def getHostGuestMapping(self):
                 self.logger.debug('Cluster of host %s is not virt_service, skipped', id)
                 continue
 
-            if self.config.hypervisor_id == 'uuid':
+            if self.config['hypervisor_id'] == 'uuid':
```
Comment (cnsnyder) [inline virtwho/virt/rhevm/rhevm.py:236]: Should this be in it's own _validate_hypervisor_id method?

**Review by cnsnyder — CHANGES_REQUESTED** [review]

File: virtwho/virt/rhevm/rhevm.py [hunk 3]
```diff
@@ -262,22 +292,10 @@ def getHostGuestMapping(self):
...
-    if len(sys.argv) < 4:
-        print("Usage: %s url username password" % sys.argv[0])
-        sys.exit(0)
-
-    import logging
-    logger = logging.Logger("")
-    config = Config('rhevm', 'rhevm', server=sys.argv[1], username=sys.argv[2],
-                    password=sys.argv[3])
-    rhevm = RhevM(logger, config)
-    print dict((host, [guest.toDict() for guest in guests]) for host, guests in rhevm.getHostGuestMapping().items())
```
Comment (jirihnidek) [inline virtwho/virt/rhevm/rhevm.py:283]: @cnsnyder Is it OK to remove this code? I consider it as obsolete.

File: tests/test_config_section_libvirtd.py [hunk 4]
```diff
@@ -102,6 +102,47 @@ def test_validate_server_full_url(self):
...
+
+    def test_validate_unvalid_hypervisor_id(self):
+        """
+        Test validation of valid hypervisor_id
+        """
+        self.init_virt_config_section()
+        self.virt_config['hypervisor_id'] = 'unsupported_id'
+        result = self.virt_config._validate_hypervisor_id('hypervisor_id')
+        expected_result = [
+            ('warning', 'Invalid option: "unsupported_id" for hypervisor_id, use one of: (uuid, hostname)')
```
Comment (cnsnyder) [inline tests/test_config_section_libvirtd.py]: This should be 'error' level as it invalidates the configuration to use an unsupported_id.
The configuration needs a valid hypervisor_id and so it should be considered invalid if a bad one is given.

**Review by cnsnyder — CHANGES_REQUESTED** [review]
Please update the message from 'warning' to 'error' for the validate_hypervisor_id method (and associated tests)

**Review by cnsnyder — APPROVED** [review]

**Comment by cnsnyder** [comment]
LGTM! Please squash, and I'll merge!

**Pull request merged**